modules/telnetd/src/main/scala/command/eph.scala

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package telnetd
+package command
+
+import gem.enum.Site
+import gem.horizons.EphemerisContext
+import gem.util.Timestamp
+
+import cats.data.NonEmptyList
+import cats.implicits._
+import com.monovore.decline.{ Command => _, _ }
+import tuco._
+import Tuco._
+import tuco.shell._
+
+import java.nio.file.Path
+import java.time.{ Instant, LocalDateTime }
+import java.time.ZoneOffset.UTC
+import java.time.format.DateTimeFormatter
+
+/** Commands for working with ephemerides.
+  *
+  * exportCommand - exports ephemerides to files for the TCS
+  * reportCommand - display ephemerides status information
+  * updateCommand - updates ephemerides from horizons if necessary
+  */
+object eph extends SiteOpt {
+
+  /** Export directory option. */
+  val dir: Opts[Path] =
+    Opts.option[Path]("dir", help = "export to directory", short = "d")
+      .validate("Directory must be writeable") { _.toFile.canWrite }
+
+  /** Ephemeris key argument parsing.
+    *
+    * @param name     name of the ephemeris key type (e.g., "horizons-key", or
+    *                 generic "ephemeris-key"
+    * @param failMsg  text to display if a key cannot be parsed
+    * @param collectf function that selects a subset of valid ephemeris keys
+    *                 (e.g., only those that are also horizons keys)
+    */
+  private def keys[K <: EphemerisKey](
+    name:    String,
+    failMsg: String => String
+  )(
+    collectf: PartialFunction[EphemerisKey, K]
+  ): Opts[NonEmptyList[K]] =
+
+    Opts.arguments[String](name)
+      .orEmpty
+      .mapValidated {
+        _.foldMap { keyStr =>
+          EphemerisKey.fromString
+            .getOption(keyStr)
+            .collect(collectf)
+            .fold(failMsg(keyStr).invalidNel[List[K]]) { k =>
+              List(k).validNel[String]
+            }
+        }.withEither(_.flatMap {
+          case Nil    => Left(NonEmptyList.one("Must supply at least one valid key"))
+          case h :: t => Right(NonEmptyList.of(h, t: _*))
+        })
+      }
+
+  /** Argument for one or more ephemeris keys of any type. */
+  val anyKeys: Opts[NonEmptyList[EphemerisKey]] =
+    keys("ephemeris-key", keyStr => s"'$keyStr' is not a valid ephemeris key") {
+      case k => k
+    }
+
+  /** Argument for one or more horizons keys. */
+  val horizonsKeys: Opts[NonEmptyList[EphemerisKey.Horizons]] =
+    keys("horizons-key", keyStr => s"'$keyStr' is not a valid horizons key") {
+      case k: EphemerisKey.Horizons => k
+    }
+
+  /** Command that exports ephemeris files in TCS format to a directory on the
+    * server.
+    */
+  val exportCommand: GemCommand =
```

Comment:
Export is kind of lame because it exports to a directory on the server.  Since there is no configuration, you have to supply said directory.  This is how ephemerides are communicated to the TCS though.  In production of course we'll want some kind of cron job that does this on a daily basis.

Reply:
I remember the old NCSA Telnet program had a feature to insert the ftp command to go back to your local machine, so this problem has been around forever. Maybe it could `scp` the file for you, so you have to give user@machine:path and it would prompt for the password. But this is fine for now.